File: apps/mobile-android/app/src/main/java/com/raweval/axis/ui/Tokens.kt
package com.raweval.axis.ui

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Axis design tokens — mirror the Tailwind palette from apps/web.
 * See docs/mobile/design.md.
 */
object AxisColors {
    // Light
    val Canvas       = Color(0xFFF7F8FA)
    val Raised       = Color(0xFFFFFFFF)
    val Subtle       = Color(0xFFEEF1F6)
    val Ink          = Color(0xFF0F172A)
    val InkSecondary = Color(0xFF475569)
    val InkTertiary  = Color(0xFF64748B)
    val Edge         = Color(0xFFE2E8F0)
    val Brand        = Color(0xFF2563EB)
    val Success      = Color(0xFF16A34A)
    val Warning      = Color(0xFFD97706)
    val Danger       = Color(0xFFDC2626)

    // Dark
    val CanvasDark       = Color(0xFF111119)
    val RaisedDark       = Color(0xFF1E1E28)
    val InkDark          = Color(0xFFEEF0F3)
    val EdgeDark         = Color(0xFF333641)
    val SubtleDark       = Color(0xFF262630)
    val InkSecondaryDark = Color(0xFFADB5C4)
}

object AxisRadius {
    val Card = 10.dp
    val Chip = 6.dp
}

object AxisSpacing {
    val Tight = 8.dp
    val Base = 12.dp
    val Loose = 16.dp
    val Section = 24.dp
}

File: apps/mobile-ios/Axis/Design/Tokens.swift
import SwiftUI

/// Axis design tokens — mirror the Tailwind palette from apps/web.
/// See docs/mobile/design.md.
enum AxisColors {
    // Light mode
    static let canvas       = Color(red: 0.969, green: 0.973, blue: 0.980)   // #F7F8FA
    static let raised       = Color.white                                    // #FFFFFF
    static let subtle       = Color(red: 0.933, green: 0.945, blue: 0.965)   // #EEF1F6
    static let ink          = Color(red: 0.059, green: 0.090, blue: 0.165)   // #0F172A
    static let inkSecondary = Color(red: 0.278, green: 0.333, blue: 0.412)   // #475569
    static let inkTertiary  = Color(red: 0.392, green: 0.455, blue: 0.545)   // #64748B
    static let edge         = Color(red: 0.886, green: 0.910, blue: 0.941)   // #E2E8F0
    static let brand        = Color(red: 0.149, green: 0.388, blue: 0.922)   // #2563EB
    static let success      = Color(red: 0.086, green: 0.639, blue: 0.290)   // #16A34A
    static let warning      = Color(red: 0.851, green: 0.467, blue: 0.024)   // #D97706
    static let danger       = Color(red: 0.863, green: 0.149, blue: 0.149)   // #DC2626

    // Dark mode — adaptive colors for SwiftUI
    static let canvasDark       = Color(red: 0.067, green: 0.067, blue: 0.098)   // #111119
    static let raisedDark       = Color(red: 0.118, green: 0.118, blue: 0.157)   // #1E1E28
    static let inkDark          = Color(red: 0.933, green: 0.941, blue: 0.953)   // #EEF0F3
    static let edgeDark         = Color(red: 0.200, green: 0.212, blue: 0.255)   // #333641

    // Adaptive helper — auto-switches on colorScheme
    static func adaptive(light: Color, dark: Color) -> Color {
        // SwiftUI Color doesn't have a built-in adaptive init, but
        // wrapping in .init(UIColor { ... }) does the trick.
        Color(uiColor: UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark
                ? UIColor(dark) : UIColor(light)
        })
    }
}

enum AxisRadius {
    static let card: CGFloat = 10
    static let chip: CGFloat = 6
}

enum AxisSpacing {
    static let tight: CGFloat = 8
    static let base: CGFloat = 12
    static let loose: CGFloat = 16
    static let section: CGFloat = 24
}
